src/customHooks/useDataStorage.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react'
+
+const localStorage = window.localStorage
+
+const storeData = (name, data) => {
+  localStorage.setItem(name, JSON.stringify(data))
+}
+
+const getStoredData = (name, defaultValue) => {
+  if (!navigator.onLine) {
+    return JSON.parse(localStorage.getItem(name))
+  } else {
+    return defaultValue
+  }
+}
+
+export const useDataStorage = (dataList, isSavable) => {
+  useEffect(() => {
+    dataList.forEach((data, i) => {
```

Comment:
Looks like `'i' is declared but its value is never read` here

Reply:
You're right that's a relic from an earlier version. Deleted.